superset/viz.py

```diff
@@ -572,6 +574,51 @@ class TableViz(BaseViz):
     is_timeseries = False
     enforce_numerical_metrics = False
 
+    def process_metrics(self) -> None:
+        """Process form data and store parsed column configs.
+           1. Determine query mode based on form_data params.
+                - Use `query_mode` if it has a valid value
+                - Set as RAW mode if `all_columns` is set
+                - Otherwise defaults to AGG mode
+           2. Determine output columns based on query mode.
+        """
+        # Verify form data first: if not specifying query mode, then cannot have both
+        # GROUP BY and RAW COLUMNS.
+        fd = self.form_data
+        if (
+            not fd.get("query_mode")
+            and fd.get("all_columns")
+            and (fd.get("groupby") or fd.get("metrics") or fd.get("percent_metrics"))
+        ):
+            raise QueryObjectValidationError(
+                _(
+                    "Choose either fields to [Group By] and [Metrics] and/or "
```

Comment:
Found this error message a little confusing. Maybe something like:

`You cannot use [Columns] controls simultaneously with a combination of [Group By]/[Metrics]/[Percentage Metrics] controls. Please choose one or the other.`

Reply:
Updated. P.S. This is for legacy API, most users would probably not see this anyway.